code/4ed_render_target.cpp

```diff
@@ -152,7 +152,22 @@ draw_rectangle_outline(Render_Target *target, Rect_f32 rect, f32 roundness, f32
         vertices[i].color = color;
         vertices[i].half_thickness = half_thickness;
     }
-    draw__write_vertices_in_current_group(target, vertices, ArrayCount(vertices));
+    
+    // NOTE simon (03/04/24): If any vertex is in the render target bounds, we draw the rectangle.
+    // It would be better to use the current clip rect, but it's not available here. We could pass
+    // it down in the function signature if necessary.
+    // This is not in the loop to try minimize the impact it could have on performance when there
+    // are a lot of vertex (we don't test duplicated vertex positions).
```

Comment:
you do have the clip_rect?
if you look at: https://github.com/4coder-community/4cc/blob/a0f12509675437453121ddf270a5529ca38f58ab/code/4ed_render_target.cpp#L101
its stored in the Render_Target `target->current_clip_box`
ALthough im curious where the code _does_ use this box if not here? or does it just use that to essentially "mask" the draw call? i.e. the GPU draws everything but only whatever is drawn _inside_ that box gets presented

Reply:
Thanks, I missed that the clip box was there.
The clip box is used in the `gl_render` function (the function that actually pushes vertices to the GPU) where it is used in `glScissor` / `RSSetScissorRects` calls.